src/main/java/com/gzs/main/RestMethods.java

```diff
@@ -10,7 +10,13 @@
 @Path("/api")
 public class RestMethods {
 
-    public RestMethods() {}
+    private ObjectMapper mapper;
+    private Term term;

```

Comment:
@DespotOfMorea  term mustn't be a class variable. Multiple requests can be executed in the same time and they will invoke methods from the same instance of this class concurrently. This will cause bugs as multiple threads will store different objects in the same variable at the same time (and will return wrong values).

Reply:
Great point,I removed term from being class variable to be local variable.
```diff
-    private Term term;
     public RestMethods() {
        mapper = new ObjectMapper();
-        term = null;
``` 
```diff
-                term = DBMethods.getTermByID(term2ID);
+                Term term = DBMethods.getTermByID(term2ID);
```